docs/doc.main

```diff
@@ -240,9 +240,11 @@
       "Combinatorics/graphs.md",
       "Combinatorics/matroids.md",
       "Combinatorics/simplicialcomplexes.md",
-      "Combinatorics/partitions.md",
-      "Combinatorics/tableaux.md",
-      "Combinatorics/schur_polynomials.md",
+      "Integer combinatorics" => [
+          "Combinatorics/IntegerCombinatorics/partitions.md",
+          "Combinatorics/IntegerCombinatorics/tableaux.md",
+          "Combinatorics/IntegerCombinatorics/schur_polynomials.md",
```

Comment:
With Schur polynomials I'm not sure, the rest yes.

Reply:
I think it is okay, they are used a lot in generating functions for counting things. I definitely support splitting them off into an `AlgebraicCombinatorics` submodule in the future, if enough stuff gets developed to justify this, it would probably be a better fit for them.

Reply:
Thank you for the review! I changed the name now. (If we ever find a better place for the Schur polynomials, we can move them.)